smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ResourceIdentifierValidator.java

```diff
@@ -36,10 +36,24 @@ public final class ResourceIdentifierValidator extends AbstractValidator {
     @Override
     public List<ValidationEvent> validate(Model model) {
         return model.shapes(ResourceShape.class)
```

Comment:
Ideally we can refactor this to now use `for (ResourceShape shape : model.getResourceShapes()) {`